File: npm/scripts/parity.mjs
// JS twin of examples/parity.rs: read domains from stdin, print one
// `suffix|registrable|known` (or `NONE`) line each, using the built bundle.
// Its output must be byte-identical to the Rust harness — see the `parity`
// CI job. Run: node scripts/parity.mjs < domains.txt
import { createInterface } from "node:readline";

import { lookup } from "../dist/index.js";

const rl = createInterface({ input: process.stdin, crlfDelay: Infinity });
const lines = [];
for await (const domain of rl) {
  const info = lookup(domain);
  lines.push(info === undefined ? "NONE" : `${info.suffix}|${info.registrableDomain ?? ""}|${info.known}`);
}
process.stdout.write(lines.join("\n") + (lines.length ? "\n" : ""));

File: examples/parity.rs
//! Parity harness: read domains from stdin, print `suffix|registrable|known`
//! (or `NONE`) per line. Paired with a JS twin to prove the TS port matches.
use std::io::{self, BufRead, Write};
use structured_public_domains::lookup;

fn main() {
    let stdin = io::stdin();
    let stdout = io::stdout();
    let mut out = stdout.lock();
    for line in stdin.lock().lines() {
        let Ok(domain) = line else { break };
        match lookup(&domain) {
            Some(info) => {
                let reg = info.registrable_domain().unwrap_or("");
                let _ = writeln!(out, "{}|{}|{}", info.suffix(), reg, info.is_known());
            }
            None => {
                let _ = writeln!(out, "NONE");
            }
        }
    }
}
